legate/core/_legion/future.py

```diff
@@ -94,10 +96,17 @@ def set_value(
         type : object
             An optional object to represent the type of the future
         """
+        if provenance is None:
+            provenance = ""
         if self.handle is not None:
             raise RuntimeError("Future must be unset to set its value")
-        self.handle = legion.legion_future_from_untyped_pointer(
-            runtime, ffi.from_buffer(data), size
+        self.handle = legion.legion_future_from_untyped_pointer_detailed(
+            runtime,
+            ffi.from_buffer(data),
+            size,
+            False,
```

Comment:
Can we instead do

```
None if provenance is None else provenance.encode()
```

in this and other newly introduced calls, to avoid passing a "fake" provenance (instead by passing NULL we explicitly tell Legion there's no provenance)

 I don't know if it even makes a difference...